### ISSUE TITLE
Deserialization of inner documents is not working

### ISSUE THREAD

**Issue by gilbsgilbs** [elastic/elasticsearch-dsl-py#810]
Hi,
Inner documents don't seem to be deserialized properly in 6.0.1. This diff adds a test case to reproduce the issue:

```diff
diff --git a/test_elasticsearch_dsl/test_integration/test_document.py b/test_elasticsearch_dsl/test_integration/test_document.py
index 324f645..b4c9ded 100644
--- a/test_elasticsearch_dsl/test_integration/test_document.py
+++ b/test_elasticsearch_dsl/test_integration/test_document.py
@@ -45,12 +45,14 @@ class Commit(DocType):
 
 class Comment(InnerDoc):
     content = Text()
+    created_at = Date()
     author = Object(User)
     class Meta:
         dynamic = MetaField(False)
 
 class PullRequest(DocType):
     comments = Nested(Comment)
+    created_at = Date()
     class Meta:
         index = 'test-prs'
 
@@ -67,7 +69,13 @@ class SerializationDoc(DocType):
 @fixture
 def pull_request(write_client):
     PullRequest.init()
-    pr = PullRequest(_id=42, comments=[Comment(content='Hello World!', author=User(name='honzakral'))])
+    pr = PullRequest(_id=42,
+                     comments=[
+                         Comment(content='Hello World!',
+                                 author=User(name='honzakral'),
+                                 created_at='2018-01-09T10:17:03.021184'),
+                     ],
+                     created_at='2018-01-09T09:17:03.021184')
     pr.save(refresh=True)
     return pr
 
@@ -110,6 +118,17 @@ def test_nested_inner_hits_are_wrapped_properly(pull_request):
     comment = pr.meta.inner_hits.comments.hits[0]
     assert isinstance(comment, Comment)
 
+
+def test_nested_inner_hits_are_deserialized_properly(pull_request):
+    s = PullRequest.search().query('nested', inner_hits={}, path='comments',
+                                   query=Q('match', comments__content='hello'))
+
+    response = s.execute()
+    pr = response.hits[0]
+    assert isinstance(pr.created_at, datetime)  # this works OK
+    assert isinstance(pr.comments[0].created_at, datetime)  # This fails, inner doc is not deserialized
+
+
 def test_nested_top_hits_are_wrapped_properly(pull_request):
     s = PullRequest.search()
     s.aggs.bucket('comments', 'nested', path='comments').metric('hits', 'top_hits', size=1)
```

The test reproduces with a nested document, but the problem is the same with inner objects.

Thanks!

**Comment by HonzaKral** [comment]
Thank you for the report, would you mind submitting a PR with the test?

**Comment by HonzaKral** [comment]
A push has been fixed, could you please verify that it works for you? Thanks again!

**Comment by gilbsgilbs** [comment]
@HonzaKral I can confirm d54acb56688aeb4727d5e32a5d60d211a78a8a30 fixes the issue.

**Comment by HonzaKral** [comment]
Released as `6.1.0`
